pom.xml

```diff
@@ -1205,6 +1200,14 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.1.2</version>
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>9.0.1</version>
```

Comment:
is this like a submodule of checkstyle? wondering why it's newly added

Reply:
9.0.1 is the version of the [checkstyle language](https://checkstyle.sourceforge.io/releasenotes.html), which is independent from the version of the [maven checkstyle plugin](http://maven.apache.org/plugins/maven-checkstyle-plugin/history.html) 3.1.2